Sources/MessagesDisplayDataSource.swift

```diff
@@ -25,41 +25,52 @@
 import Foundation
 
 public protocol MessagesDisplayDataSource: class, MessagesDataSource {
-
-    func messageColor(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> UIColor
-
-    func avatar(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> Avatar
-
+    
+    func textColor(for message: MessageType, at indexPath: IndexPath) -> UIColor
+    
+    func backgroundColor(for message: MessageType, at  indexPath: IndexPath) -> UIColor
+    
+    func avatar(for message: MessageType) -> Avatar
```

Comment:
Sorry, I meant we should continue to use:
```Swift
func avatar(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> Avatar
```
The goal here is to be able to access any MessageKit component from these methods, just like how UIKit does with UICollectionView & UITableView.

We can access the specific row, the specific section, the `MessagesCollectionView` itself, thus we can also access the `MessagesCollectionViewFlowLayout` object, you can even access a specific `MessagesCollectionViewCell`. I'm not looking for brevity but rather flexibility and customization. @fbartho already mentioned he would need access to more components than just the `MessageType` to make his customizations